engine/config/yaml_config.cc

```diff
@@ -266,6 +268,9 @@ void YamlHandler::UpdateModelConfig(ModelConfig new_model_config) {
     if (!model_config_.grammar.empty())
       yaml_node_["grammar"] = model_config_.grammar;
 
+    if (!std::isnan(static_cast<double>(model_config_.size)))
```

Comment:
Shouldn't this be `static_cast<uint64_t>`?

Reply:
Nice catch. I reused all the checks above, even for int, and they all double cast check.

Reply:
Also, uint64_t can never represent a nan value. Isn't it?

Reply:
Removed as confirmed